packages/xstate-vue/src/useMachine.ts

```diff
@@ -1,59 +1,29 @@
-import { Ref, shallowRef } from 'vue';
 import {
+  ActorRefFrom,
   AnyStateMachine,
   AreAllImplementationsAssumedToBeProvided,
-  InternalMachineImplementations,
-  InterpreterFrom,
+  EventFrom,
   InterpreterOptions,
-  StateFrom
+  MissingImplementationsError,
+  SnapshotFrom
 } from 'xstate';
-import { MaybeLazy, Prop } from './types.ts';
-import { useInterpret } from './useInterpret.ts';
-
-type RestParams<TMachine extends AnyStateMachine> =
-  AreAllImplementationsAssumedToBeProvided<
-    TMachine['__TResolvedTypesMeta']
-  > extends false
-    ? [
-        options: InterpreterOptions<TMachine> &
-          InternalMachineImplementations<
-            TMachine['__TContext'],
-            TMachine['__TEvent'],
-            TMachine['__TResolvedTypesMeta'],
-            true
-          >
-      ]
-    : [
-        options?: InterpreterOptions<TMachine> &
-          InternalMachineImplementations<
-            TMachine['__TContext'],
-            TMachine['__TEvent'],
-            TMachine['__TResolvedTypesMeta']
-          >
-      ];
-
-type UseMachineReturn<
-  TMachine extends AnyStateMachine,
-  TInterpreter = InterpreterFrom<TMachine>
-> = {
-  state: Ref<StateFrom<TMachine>>;
-  send: Prop<TInterpreter, 'send'>;
-  service: TInterpreter;
-};
+import { useActor } from './useActor.ts';
+import { Ref } from 'vue';
 
+/**
+ * @deprecated Use `useActor(...)` instead.
+ */
 export function useMachine<TMachine extends AnyStateMachine>(
```

Comment:
I wonder how useful it really is to keep this around if its signature changed *anyway*. Right now, the second argument only accepts `InterpreterOptions` whereas previously it also accepted `InternalMachineImplementations`.